### ISSUE TITLE
Generating changelog from commits is erratic

### ISSUE THREAD

**Issue by JuergenSimon** [github-tools/github-release-notes#219]
Consider the following commit history:

```
8d8e335 (HEAD -> master, tag: v1.0.32, origin/master, origin/HEAD) [bug] Fixed foo. 
cd653e0 (tag: v1.0.31) [bug] Fixed foo.
993744e [bug] Fixed bar. 
ae9de2a (tag: v1.0.30) [enhancement] Added Foo
8f246ec [bug] Fix foo. [enhancement] Added Bar. 
d3a2f49 (origin/dev, dev) [iirn] Linting. 
e5d2ae4 [iirn] Fix tests.
a7cbde4 [iirn] Package updates
06a80c9 [enhancement] Added FooBar
7a8ac68 (tag: v1.0.29) Merge pull request #65
30ae066 [enhancement] Added FooBar
e32d394 [iirn] Add gren scripts. 
```

Given the following call:

```
gren changelog --generate --data-source commits --tags=all --override --ignore-commits-with \[iirn\]
```

I would expect the changelog to look something like:

```
v1.0.32
8d8e335 (HEAD -> master, tag: v1.0.32, origin/master, origin/HEAD) [bug] Fixed foo. 

v1.0.31
cd653e0 (tag: v1.0.31) [bug] Fixed foo.
993744e [bug] Fixed bar. 

v1.0.30
ae9de2a (tag: v1.0.30) [enhancement] Added Foo
8f246ec [bug] Fix foo. [enhancement] Added Bar. 
06a80c9 [enhancement] Added FooBar

v1.0.29
7a8ac68 (tag: v1.0.29) Merge pull request #65
30ae066 [enhancement] Added FooBar
e32d394 [iirn] Add gren scripts. 
...
```

Instead I am getting:

```
v1.0.32
(empty)

v1.0.31
(empty)

v1.0.30
06a80c9 [enhancement] Added FooBar

v1.0.29
30ae066 [enhancement] Added FooBar
```

There are many commit messages missing. Seems to me every commit message at a tagged position is missing and then some.

**Comment by blairg23** [comment]
I am also having this problem, has this been triaged yet?

**Comment by alexcanessa** [comment]
This should be fixed by 2460facf950c3351a6cd7f642ffac10f4b965388
